op-node/rollup/driver/state.go

```diff
@@ -443,3 +456,49 @@ type hashAndErrorChannel struct {
 	hash common.Hash
 	err  chan error
 }
+
+// checkForGapInUnsafeQueue checks if there is a gap in the unsafe queue and attempts to retrieve the missing payloads from the backup RPC.
+// WARNING: This function fails silently (aside from warning logs).
+func (s *Driver) checkForGapInUnsafeQueue(ctx context.Context) {
+	start, end := s.derivation.GetUnsafeQueueGap()
+	size := end - start
+
+	// If there is a gap in the queue and a backup sync client is configured, attempt to retrieve the missing payloads from the backup RPC
+	if size > 0 && s.l2SyncCl != nil {
+		// Attempt to fetch the missing payloads from the backup unsafe sync RPC concurrently.
+		// Concurrent requests are safe here due to the engine queue being a priority queue.
+		// TODO: Should enforce a max gap size to prevent spamming the backup RPC or being rate limited.
+		for blockNumber := start; blockNumber < end; blockNumber++ {
+			go s.fetchUnsafeBlockFromRpc(ctx, blockNumber)
+		}
+	}
+}
+
+// fetchUnsafeBlockFromRpc attempts to fetch an unsafe execution payload from the backup unsafe sync RPC.
+// WARNING: This function fails silently (aside from warning logs).
+func (s *Driver) fetchUnsafeBlockFromRpc(ctx context.Context, blockNumber uint64) {
+	s.log.Info("requesting unsafe payload from backup RPC", "block number", blockNumber)
+
+	// TODO: Post Shanghai hardfork, the engine API's `PayloadBodiesByRange` method will be much more efficient, but for now,
```

Comment:
@protolambda does this imply that we'll need to include engine authentication when we do this update?

Reply:
Yes, we can make it optional though: e.g. "to use an engine API endpoint for faster sync specify the `--sync.auth` flag" and we'll authenticate the RPC like an engine RPC. For now we can just stick to regular unauthenticated JSON-RPC.